### PR TITLE
Rename the CLI for local Cilium API access to 'cilium-dbg'

### PR DESCRIPTION
**Pull request by joestringer**
Requires: https://github.com/cilium/cilium-cli/pull/2002
Requires: https://github.com/cilium/cilium/pull/28406
Requires: https://github.com/cilium/cilium-cli/pull/2006

- cilium: Rename local CLI to cilium-dbg                                                                                                                                                                                     
- bugtool: Update commands for renamed cilium-dbg                                                                                                                                                           
- docs: Update cmdref for renamed cilium-dbg                                                                                                                                                                
- docs: Update example commands for cilium-dbg rename                                                                                                                                              
- .github: Update CLI calls for renamed cilium-dbg                                                                                                                                                          
- test: Update bash scripts for renamed cilium-dbg                                                                                                                                                          
- test: Update Go tests for renamed cilium-dbg                                                                                                                                                              
- images: Switch CLI references to cilium-dbg

### CODE REVIEW

File: cilium/.gitignore [hunk 1]
```diff
@@ -1 +0,0 @@
-cilium
```
Comment (nebril) [inline cilium/.gitignore:1]: Shouldn't this be changed to `cilium-dbg` too?

**Review by nebril — APPROVED** [review]
Holy mother of renaming! LGTM, found one .gitignore issue, but it's non-blocking.

File: cilium-dbg/cmd/root.go [hunk 2]
```diff
@@ -93,14 +93,14 @@ const completionExample = `
 ## or, if running Bash 4.1+
 	brew install bash-completion@2
 ## afterwards you only need to run
-	cilium completion bash > $(brew --prefix)/etc/bash_completion.d/cilium
+	cilium-dbg completion bash > $(brew --prefix)/etc/bash_completion.d/cilium
```
Comment (ti-mo) [inline cilium-dbg/cmd/root.go:96]: Nit: Under which circumstances would someone run cilium-dbg on macOS? :thinking: It's technically possible, just wondering if we encourage this or not.

**Review by ti-mo — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Review by aspsk — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Review by zacharysarah — APPROVED** [review]
LGTM 👍🏻

**Review by julianwiedmann — APPROVED** [review]

File: cilium-dbg/cmd/root.go (hunk 2 above)
Comment (joestringer) [reply cilium-dbg/cmd/root.go:96]: From a quick perusal of git history, it seems like this was a boilerplate output added way back in 2017. I agree with you, I don't see a path to use this tool on macOS. We could probably remove this bit. Filed https://github.com/cilium/cilium/issues/28446 to follow up.

File: cilium/.gitignore (hunk 1 above)
Comment (joestringer) [reply cilium/.gitignore:1]: Good catch. I'll fix this up when I submit the final version with passing external workloads test :pray:

**Review by gentoo-root — APPROVED** [review]
Nice, no more confusion between cilium-cli and agent CLI.

**Comment by joestringer** [comment]
Hey @YutaroHayakawa , any chance of a review for your codeowners?

**Review by YutaroHayakawa — APPROVED** [review]

**Comment by YutaroHayakawa** [comment]
Sorry for the late. Looks good to me!

**Pull request merged**